File: app/src/main/java/com/example/mrakopediareader/api/parser/CategoryParser.java
package com.example.mrakopediareader.api.parser;

import com.example.mrakopediareader.api.dto.Category;

import org.json.JSONException;
import org.json.JSONObject;

public class CategoryParser extends ParserImpl<Category> {
    private static String KEY_TITLE = "title";
    private static String KEY_URL = "url";

    @Override
    public Category fromJsonObject(JSONObject object) throws JSONException {
        final String title = object.getString(KEY_TITLE);
        final String url = object.getString(KEY_URL);
        return new Category(title, url);
    }
}

File: app/src/main/java/com/example/mrakopediareader/api/parser/CategoryParser.kt
package com.example.mrakopediareader.api.parser

import com.example.mrakopediareader.api.dto.Category
import org.json.JSONObject

internal class CategoryParser : ParserImpl<Category>() {
    override fun fromJsonObject(jsonObject: JSONObject): Category {
        val title = jsonObject.getString(KEY_TITLE)
        val url = jsonObject.getString(KEY_URL)
        return Category(title, url)
    }

    companion object {
        private const val KEY_TITLE = "title"
        private const val KEY_URL = "url"
    }
}